stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -917,17 +931,30 @@ extension ${Self}: BinaryFloatingPoint {
   /// - If `x` is `greatestFiniteMagnitude`, then `x.nextUp` is `infinity`.
   @_inlineable // FIXME(sil-serialize-all)
   public var nextUp: ${Self} {
-    if isNaN { return self }
-    if sign == .minus {
+%if bits == 32 or bits == 64:
+    // Silence signaling NaNs, map -0 to +0.
+    let x = self + 0
 #if arch(arm)
-      // On arm, subnormals are flushed to zero.
-      if (exponentBitPattern == 1 && significandBitPattern == 0) ||
-         (exponentBitPattern == 0 && significandBitPattern != 0) {
-        return ${Self}(sign: .minus,
-          exponentBitPattern: 0,
-          significandBitPattern: 0)
+    // On arm, skip positive subnormal values.
+    if _slowPath(x.bitPattern & (-${Self}.infinity).bitPattern == 0) {
+      return .leastNormalMagnitude
+    }
+#endif
+    if _fastPath(x < .infinity) {
+      let increment = Int${bits}(bitPattern: x.bitPattern) &>> ${bits - 1} | 1
+      let bitPattern_ = x.bitPattern &+ UInt${bits}(bitPattern: increment)
+#if arch(arm)
+      // On arm, flush negative subnormal values to -0.
```

Comment:
If you follow my suggestion above, you can eliminate this, since all subnormals will have already been handled.

Reply:
The subnormal result of incrementing `-.leastNormalMagnitude` still needs to be flushed to -0, but that can be written differently and in a way that is future-proof.

Reply:
It should fall out for free. `1` will be subtracted from the bit pattern, which will produce a negative subnormal, which will be treated as zero by whatever consumes it (as a "non canonical encoding of zero"). Or am I missing something?

Reply:
Well, it's defensible since it's technically correct, but I think it'd be rather more elegant not to _produce_ the negative subnormal in the first place; the end user may inspect its bit pattern and we might as well give a canonical encoding of negative zero because we're nice like that. The cost to doing so would be minimal and it can be future-proof too.

Reply:
ok.